File: backend/BusinessOS/src/main/java/com/businessos/modules/finance/reports/service/FinancialReportService.java
package com.businessos.modules.finance.reports.service;

import com.businessos.modules.finance.reports.dto.AccountLedger;
import com.businessos.modules.finance.reports.dto.AgeingReport;
import com.businessos.modules.finance.reports.dto.BalanceSheetReport;
import com.businessos.modules.finance.reports.dto.CashFlowReport;
import com.businessos.modules.finance.reports.dto.ProfitLossReport;
import com.businessos.modules.finance.reports.dto.TrialBalanceReport;

import java.time.LocalDate;

public interface FinancialReportService {
    ProfitLossReport generateProfitLossReport(LocalDate startDate, LocalDate endDate);
    BalanceSheetReport generateBalanceSheetReport(LocalDate asOfDate);
    TrialBalanceReport generateTrialBalanceReport(LocalDate asOfDate);
    AccountLedger generateAccountLedger(Long accountId, LocalDate startDate, LocalDate endDate);
    AgeingReport generateAgeingReport(LocalDate asOfDate);
    CashFlowReport generateCashFlowReport(LocalDate startDate, LocalDate endDate);
}

File: frontend/ZuhooCMS/src/app/modules/finance/services/financial-report.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { ProfitLossReport, BalanceSheetReport, TrialBalanceReport, AgeingReport, CashFlowReport, AccountLedgerReport } from '../models/finance.model';

@Injectable({ providedIn: 'root' })
export class FinancialReportService {
  private readonly endpoint = '/company/finance/reports';
  constructor(private api: ApiService) {}

  profitLoss(startDate: string, endDate: string): Observable<ProfitLossReport> {
    return this.api.get<ProfitLossReport>(`${this.endpoint}/profit-loss`, { startDate, endDate });
  }

  balanceSheet(asOfDate: string): Observable<BalanceSheetReport> {
    return this.api.get<BalanceSheetReport>(`${this.endpoint}/balance-sheet`, { asOfDate });
  }

  trialBalance(asOfDate: string): Observable<TrialBalanceReport> {
    return this.api.get<TrialBalanceReport>(`${this.endpoint}/trial-balance`, { asOfDate });
  }

  ageing(asOfDate: string): Observable<AgeingReport> {
    return this.api.get<AgeingReport>(`${this.endpoint}/ageing`, { asOfDate });
  }

  cashFlow(startDate: string, endDate: string): Observable<CashFlowReport> {
    return this.api.get<CashFlowReport>(`${this.endpoint}/cash-flow`, { startDate, endDate });
  }

  accountLedger(accountId: number, startDate: string, endDate: string): Observable<AccountLedgerReport> {
    return this.api.get<AccountLedgerReport>(`${this.endpoint}/ledger`, { accountId, startDate, endDate });
  }
}
